channel_test.go

```diff
@@ -0,0 +1,181 @@
+package p9p
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"net"
+	"testing"
+	"time"
+)
+
+// TestWriteOverflow ensures that a Twrite message will have the data field
+// truncated if the msize would be exceeded.
+func TestWriteOverflow(t *testing.T) {
+	const (
+		msize         = 512
+		overflowMSize = msize * 3 / 2
```

Comment:
I'd like to cover the case where len(p) fits in msize, but len(p) + [size of envelope] don't to the regression suite. That would avoid some potential naïve (and wrong) optimizations.